specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,267 @@
+                            CHECK_FAILURE(args->put_SaveAsType(dialog.selectedType));
+                            CHECK_FAILURE(args->put_AllowReplace(dialog.allowReplace));
+
+                            // Confirm to download, required
+                            CHECK_FAILURE(args->put_ConfirmToSave(TRUE));
+                        }
+                    };
+
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    m_appWindow->RunAsync(
+                        [deferral, showCustomizedDialog]()
+                        {
+                            showCustomizedDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_saveAsRequestedToken);
+    }
+    m_silentSaveAs = !m_silentSaveAs;
+    MessageBox(
+        m_appWindow->GetMainWindow(),
+        (m_silentSaveAs ? L"Silent Save As Enabled" : L"Silent Save As Disabled"), L"Info",
+        MB_OK);
+    return true;
+}
+//! [ToggleSilent]
+
+//! [ProgrammaticSaveAs]
+// Call SaveContentAs method to trigger the programmatic save as.
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2Staging20)
+        return false;
+    m_webView2Staging20->SaveContentAs(
+        Callback<ICoreWebView2StagingSaveContentAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS result) -> HRESULT
+            {
+                // Show SaveContentAs returned result, optional
+                // Skip message box when the result is COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG (0),
+                // to prevent it deactivates save as modal dialog focus on the window
+                if (result > 0)
+                {
+                    MessageBox(
+                        m_appWindow->GetMainWindow(),
+                        (L"Save As " + saveAsResultString[result]).c_str(), L"Info", MB_OK);
+                }
+                return S_OK;
+            })
+            .Get());
+    return true;
+}
+//! [ProgrammaticSaveAs]
+```
+# API Details
+## Win32 C++
+```c++
+/// Specifies save as type selection options for `ICoreWebView2Staging20`,
+/// used in `SaveAsRequestedEventArgs`
+///
+/// When the source is a html page, allows to select `HTML_ONLY`,
+/// `SINGLE_FILE`, `COMPLETE`; when the source is a non-html,
+/// only allows to select `DEFAULT`; otherwise, will deny the download
+/// and return `COREWEBVIEW2_SAVE_AS_TYPE_NOT_SUPPORTED`.
+///
+/// The content type/format is a MIME type, indicated by the source
+/// server side and identified by the browser. It’s not related to the
+/// file’s type or extension. MIME type of `text/html`,
+/// `application/xhtml+xml` are considered as html page.
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_AS_TYPE {
```

Comment:
We need to rename this. Generally we don't use the word Type because it is overloaded with programming language types and in this case is also confusing with content-types or mime-types. Generally we use the word Kind instead of Type. For this case we should also add an additional word to help better describe what this enum actually is. 

Its about how to save a document that consists of multiple files? How about COREWEBVIEW2_MULTIPLE_FILE_DOCUMENT_SAVE_AS_KIND?

Reply:
In my reading the prefix is COREWEBVIEW2 is the product, SAVE_CONTENT_AS is the feature, and then what's not the prefix is just KIND which doesn't really say anything about what this enum represents. We can see if reviewers have any other suggestions.